packages/core/src/account/base.ts

```diff
@@ -30,11 +31,28 @@ export enum DeploymentState {
 export interface BaseSmartAccountParams<
   TTransport extends SupportedTransports = Transport
 > {
+  /* Required */
   rpcClient: string | PublicErc4337Client<TTransport>;
-  entryPointAddress: Address;
   factoryAddress: Address;
-  owner?: SmartAccountSigner | undefined;
   chain: Chain;
+
+  /* Optional */
+
```

Comment:
nit: don't think we need to distinguish at the interface. the code speaks for itself :)
```suggestion
```